klippy/extras/fan.py

```diff
@@ -54,8 +69,15 @@ def set_speed_from_command(self, value):
                                               self.set_speed(pt, value)))
     def _handle_request_restart(self, print_time):
         self.set_speed(print_time, 0.)
+
+    def tach_callback(self, time, freq):
+        self.rpm = freq * 30. / self.tach_ppr # 2 ppr encoder
+
     def get_status(self, eventtime):
-        return {'speed': self.last_fan_value}
+        return {
+            'speed': self.last_fan_value,
+            'rpm': self.rpm,
```

Comment:
FWIW, I think it would be preferable to try and avoid the use of callbacks here.  The callback is run in a background thread, and many developers may not realize that.  I think it would be preferable if the main class that calls `mcu.register_response()` be the only class to deal with the multi-threading issues.  (That is, it can obtain the values, store them internally, and make them available on request.  Ideally, that would allow the calculations to be done entirely from the main thread and limit the chance of multi-threading issues.